gradle-plugin-testing-junit/src/main/java/com/palantir/gradle/testing/files/gradle/PluginsBlock.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.testing.files.gradle;
+
+import com.google.errorprone.annotations.RestrictedApi;
+import com.palantir.gradle.testing.RestrictedCreation;
+import com.palantir.gradle.testing.files.ProjectFile.FileEditor;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.intellij.lang.annotations.Language;
+
+/**
+ * Structured API for managing the {@code plugins {}} block in Gradle files.
+ * <p>
+ * Example: {@code rootProject.buildGradle().plugins().add("java").add("application")}
+ *
+ * @see StructuredGradleFile#plugins()
+ */
+public record PluginsBlock(Function<FileEditor, ? extends GradleFile> editorFunction) {
+    private static final Pattern PLUGINS_BLOCK_PATTERN = Pattern.compile("plugins\\s*\\{([^}]*)}\\n?", Pattern.DOTALL);
+    private static final Pattern BUILDSCRIPT_BLOCK_PATTERN =
+            Pattern.compile("buildscript\\s*\\{(?:[^{}]|\\{[^{}]*})*}", Pattern.DOTALL);
+
+    @RestrictedApi(explanation = RestrictedCreation.EXPLANATION, allowedOnPath = RestrictedCreation.ALLOWED_ON_PATH)
+    public PluginsBlock {}
+
+    /**
+     * Adds a plugin without a version. Example: {@code add("java")}
+     */
+    public PluginsBlock add(@Language("Gradle") String pluginId) {
+        return addPlugin(new Plugin(pluginId, Optional.empty(), Optional.empty()));
+    }
+
+    /**
+     * Adds a plugin with a version. Example: {@code add("com.example.plugin", "1.0.0")}
+     */
+    public PluginsBlock add(@Language("Gradle") String pluginId, String version) {
```

Comment:
I actually think we don't want to add this one, as it will be forever stuck at a fixed version. Ideally, any time we let people add external plugin versions they should do it in a way that enables it to be upgraded automatically. Either the `gradlePluginForTesting` approach (which I'm still not 100% sure about) or we don't let people give a version like this, instead reading the version from a file that can be upgraded by excavator.